sklearn/feature_selection/_univariate_selection.py

```diff
@@ -297,15 +297,30 @@ def f_regression(X, y, *, center=True):
     else:
         X_norms = row_norms(X.T)
 
+    if np.any(X_norms == 0) or np.ptp(y) == 0:
+        # The Pearson correlation coefficient is undefined for random
+        # variables (attributes and target) with a standard deviation
+        # of zero
+        raise ValueError("The correlation is undefined for features "
+                         "or target with identically constant values.")
+
     # compute the correlation
     corr = safe_sparse_dot(y, X)
     corr /= X_norms
     corr /= np.linalg.norm(y)
 
-    # convert to p-value
     degrees_of_freedom = y.size - (2 if center else 1)
-    F = corr ** 2 / (1 - corr ** 2) * degrees_of_freedom
-    pv = stats.f.sf(F, 1, degrees_of_freedom)
+
+    # The degrees of freedom must be at least one for the F-test
+    if degrees_of_freedom < 1:
+        raise ValueError("The sample size must be greater than two if "
+                         f"'center=True'. Got n_samples={n_samples}.")
```

Comment:
If `center=True`, then `degrees_of_freedom=n_samples - 2` ->  `n_samples` must be greater than 3?

And with the same logic, `center=False`, -> `n_samples` must be greater than 2?

Reply:
> If `center=True`, then `degrees_of_freedom=n_samples - 2` -> `n_samples` must be greater than 3?
> 

IIRC, `degrees_of_freedom` must be greater than 0 (or greater than or equal 1). Therefore, if `center=True`, then,
`n_samples` must be greater than 2 (or greater than or equal 3).

> And with the same logic, `center=False`, -> `n_samples` must be greater than 2?

Applying the same logic, if `center=False`, then, `n_samples` must be greater than 1 (or greater than or equal 2).

I have not considered `center=False` and `n_samples=2` in the `ValueError` because this error:

https://github.com/alfaro96/scikit-learn/blob/caca0475d6483f7ff4efeab11e926f49aa03c122/sklearn/feature_selection/_univariate_selection.py#L301-L305

is raised instead.

Do you think that greater than or equal should be used instead of greater than?